dev/TabView/TabView.cpp

```diff
@@ -0,0 +1,248 @@
+            }
+        }
+    }
+
+    UpdateTabWidths();
+
+    __super::OnItemsChanged(item);
+}
+
+void TabView::OnSelectionChanged(const winrt::IInspectable& sender, const winrt::SelectionChangedEventArgs& args)
+{
+    if (m_indexToSelectOnSelectionChanged)
+    {
+        SelectedItem(Items().GetAt(m_indexToSelectOnSelectionChanged.value()));
+        m_indexToSelectOnSelectionChanged = {};
+    }
+
+    UpdateTabContent();
+}
+
+void TabView::UpdateTabContent()
+{
+    if (auto tabContentPresenter = m_tabContentPresenter.get())
+    {
+        if (!SelectedItem())
+        {
+            tabContentPresenter.Content(nullptr);
+            tabContentPresenter.ContentTemplate(nullptr);
+        }
+        else
+        {
+            auto container = ContainerFromItem(SelectedItem()).as<winrt::ListViewItem>();
+            if (container)
+            {
+                tabContentPresenter.Content(container.Content());
+                tabContentPresenter.ContentTemplate(container.ContentTemplate());
+            }
+        }
+    }
+}
+
+void TabView::CloseTab(winrt::TabViewItem const& container)
+{
+    if (auto item = ItemFromContainer(container))
+    {
+        uint32_t index = 0;
+        if (Items().IndexOf(item, index))
+        {
+            auto args = winrt::make_self<TabViewTabClosingEventArgs>(item);
+
+            m_tabClosingEventSource(*this, *args);
+
+            if (!args->Cancel())
+            {
+                Items().RemoveAt(index);
+            }
+        }
+    }
+}
+
+void TabView::OnScrollDecreaseClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        scrollViewer.ChangeView(std::max(0.0, scrollViewer.HorizontalOffset() - c_scrollAmount), nullptr, nullptr);
+    }
+}
+
+void TabView::OnScrollIncreaseClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (auto scrollViewer = m_scrollViewer.get())
+    {
+        scrollViewer.ChangeView(std::min(scrollViewer.ScrollableWidth(), scrollViewer.HorizontalOffset() + c_scrollAmount), nullptr, nullptr);
+    }
+}
+
+void TabView::UpdateTabWidths()
+{
+    double tabWidth = DoubleUtil::NaN;
+
+    if (TabWidthMode() == winrt::TabViewWidthMode::Fixed)
+    {
+        // Tabs should all be the same size, proportional to the amount of space.
+        double minTabWidth = unbox_value<double>(SharedHelpers::FindResource(c_tabViewItemMinWidthName, winrt::Application::Current().Resources(), box_value(c_tabMinimumWidth)));
+        double maxTabWidth = unbox_value<double>(SharedHelpers::FindResource(c_tabViewItemMaxWidthName, winrt::Application::Current().Resources(), box_value(c_tabMaximumWidth)));
+
+        if (auto scrollViewer = m_scrollViewer.get())
+        {
+            // Calculate the proportional width of each tab given the width of the ScrollViewer.
+            double padding = Padding().Left + Padding().Right;
```

Comment:
suggest:
```
auto padding = [auto padding = Padding()]() {return padding.Left + padding.Right}();
```

to avoid calling Padding() twice while limiting the scope of the local var we use to replace the second call.

Reply:
I find that less readable, honestly.

Reply:
I agree, but accessing the the Dependency property via the public Padding() api is not free and we should avoid calling it twice.